lib/tasks/search.rake

```diff
@@ -45,34 +45,86 @@ namespace :search do
       }
     }
 
-    Dataset.__elasticsearch__.client.indices.delete index: Dataset.__elasticsearch__.index_name rescue nil
-    Dataset.__elasticsearch__.client.indices.create index: Dataset.__elasticsearch__.index_name, body: {mappings: datasetMappings}
+    @date ||= Time.now.strftime '%Y%m%d%H%M%S'
+    @client = Dataset.__elasticsearch__.client
+    @new_index_name = "#{Dataset.index_name}_#{@date}"
+    @alias = Dataset.index_name
 
-    nb_dataset_processed = 0
-    Dataset.published.find_in_batches(batch_size: 50) do |datasets|
-      puts " Batching #{datasets.length} datasets (done #{nb_dataset_processed} / #{nb_datasets_to_index})"
-      bulk_index(datasets)
-      nb_dataset_processed = nb_dataset_processed + 50
+    def bulk_index(datasets, new_index_name)
+      begin
+        Dataset.__elasticsearch__.client.bulk({
+          index: new_index_name,
+          type: ::Dataset.__elasticsearch__.document_type,
+          body: prepare_records(datasets)
+        })
+      rescue => e
+        puts "There was an error uploading datasets:"
+        puts e
+      end
     end
-  end
 
-  def bulk_index(datasets)
-    begin
-      Dataset.__elasticsearch__.client.bulk({
-                                              index: ::Dataset.__elasticsearch__.index_name,
-                                              type: ::Dataset.__elasticsearch__.document_type,
-                                              body: prepare_records(datasets)
-                                            })
-    rescue => e
-      puts "There was an error uploading datasets:"
-      puts e
+    def prepare_records(datasets)
+      datasets.map do |dataset|
+        {index: {_id: dataset.id, data: dataset.as_indexed_json}}
+      end
     end
-  end
 
-  def prepare_records(datasets)
-    datasets.map do |dataset|
-      {index: {_id: dataset.id, data: dataset.as_indexed_json}}
+
+    def index_datasets
```

Comment:
would be better to pass function arguments that use instance vars imo

Reply:
*than use

Reply:
👍